lib/internal/promisify.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+module.exports = promisify;
+
+function promisify(fn, single) {
+  Object.defineProperty(inner, 'name', {
+    value: fn.name + 'Promisified'
+  });
+  return inner;
+
+  function inner() {
+    if (typeof arguments[arguments.length - 1] === 'function') {
+      return fn.apply(this, arguments);
+    }
+    var resolve = null;
+    var reject = null;
+    const args = Array.from(arguments);
```

Comment:
> Seems like it's not an issue anymore with the latest V8

That's why I said "compilers change over time" (but, as far as I remember, they co-exist now and try-catch will be fast only under certain conditions). Anyway, great to see progress - and with more engines coming, I'm pretty sure we don't need to care neither about deopts nor about "we will slow down callbacks if check whether they're passed or not" - this is I/O after all.